### PR TITLE
Add Support for Custom Metadata Tags

### PR DESCRIPTION
**Pull request by tonyswu**
## Proposed changes

Add support for metadata / tags in the splunk events. This is particularly useful to define the source cluster with external information that may not be available in the cluster itself, such as the cloud account id, region, etc.

An issue was submitted previously, and can be found here:

https://github.com/splunk/splunk-connect-for-kubernetes/issues/166

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](https://github.com/splunk/splunk-connect-for-kubernetes/blob/develop/CONTRIBUTING.md) doc
- [x] I have read the [CLA](https://github.com/splunk/splunk-connect-for-kubernetes/blob/develop/CLA.md)
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: helm-chart/splunk-kubernetes-logging/templates/_helpers.tpl [hunk 1]
```diff
@@ -112,6 +112,11 @@ def extract_container_info:
   | .container_name = ($cparts[:-1] | join("-"))
   | .container_id = ($cparts[-1] | rtrimstr(".log"))
   | .cluster_name = "{{ or .Values.kubernetes.clusterName .Values.global.kubernetes.clusterName | default "cluster_name" }}"
+  {{- if .Values.custom_metadata }}
+  {{-   range .Values.custom_metadata }}
```
Comment (chaitanyaphalak) [inline helm-chart/splunk-kubernetes-logging/templates/_helpers.tpl]: Use tab to indent here

File: helm-chart/splunk-kubernetes-logging/templates/_helpers.tpl [hunk 2]
```diff
@@ -112,6 +112,11 @@ def extract_container_info:
...
+  | .{{ .name }} = "{{ .value }}"
+  {{-   end }}
```
Comment (chaitanyaphalak) [inline helm-chart/splunk-kubernetes-logging/templates/_helpers.tpl]: remove extra space here

File: helm-chart/splunk-kubernetes-logging/templates/configMap.yaml [hunk 3]
```diff
@@ -236,6 +236,11 @@ data:
           {{- if or .Values.kubernetes.clusterName .Values.global.kubernetes.clusterName }}
           cluster_name
           {{- end }}
+          {{- if .Values.custom_metadata }}
+          {{-   range .Values.custom_metadata }}
```
Comment (chaitanyaphalak) [inline helm-chart/splunk-kubernetes-logging/templates/configMap.yaml]: Use tab to indent here

File: helm-chart/splunk-kubernetes-logging/templates/configMap.yaml [hunk 4]
```diff
@@ -236,6 +236,11 @@ data:
...
+          {{ .name }}
+          {{-   end }}
```
Comment (chaitanyaphalak) [inline helm-chart/splunk-kubernetes-logging/templates/configMap.yaml]: remove extra space here

**Comment by tonyswu** [comment]
Hi @chaitanyaphalak,

1. Resolved merge conflict.
2. Removed the extra space.
3. Not sure what you meant by using tab for indent. Existing indent is done with space (at least as far as I can see).

**Comment by chaitanyaphalak** [comment]
Thanks @tonyswu LGTM

**Comment by sspreitzer** [comment]
Does this also include Kubernetes labels and annotations of a Pod into an event?

**Comment by tonyswu** [comment]
No, just custom metadata that you can define manually.

**Pull request merged**